compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -1351,22 +1361,42 @@ execRepl dars importPkgs mbLedgerConfig mbAuthToken mbAppId mbSslConf mbMaxInbou
                     (Repl.runRepl importPkgs opts replHandle replLogger)
 
 -- | Remove any build artifacts if they exist.
-execClean :: ProjectOpts -> Command
-execClean projectOpts =
+execClean :: ProjectOpts -> EnableMultiPackage -> MultiPackageLocation -> MultiPackageCleanAll -> Command
+execClean projectOpts enableMultiPackage multiPackageLocation cleanAll =
   Command Clean (Just projectOpts) effect
-  where effect = do
-            withProjectRoot' projectOpts $ \_relativize -> do
-                isProject <- doesFileExist projectConfigName
-                when isProject $ do
-                    let removeAndWarn path = do
-                            whenM (doesDirectoryExist path) $ do
-                                putStrLn ("Removing directory " <> path)
-                                removePathForcibly path
-                            whenM (doesFileExist path) $ do
-                                putStrLn ("Removing file " <> path)
-                                removePathForcibly path
-                    removeAndWarn damlArtifactDir
-                    putStrLn "Removed build artifacts."
+  where
+    effect =
+      case (getEnableMultiPackage enableMultiPackage, getMultiPackageCleanAll cleanAll) of
+        (True, True) -> do
+          mMultiPackagePath <- getMultiPackagePath multiPackageLocation
+          case mMultiPackagePath of
+            Nothing -> do
+              hPutStrLn stderr "Couldn't find multi-package.yaml. Use --multi-package-path PATH or --multi-package-search to specify its location."
```

Comment:
Consider aligning and sharing this message with the one in `execBuild`.

Reply:
Ah shoot - my page didn't refresh from yesterday, i completely missed this